File: 6주차/LeetCode_179_Largest_Number/dongjun.java
/**
 *  @문제접근
 *  nums를 문자열로 변환하고
 *  람다식으로 정렬조건을 변환해 풀었습니다.
 *
 *  @성능
 *  Runtime: 5 ms
 *  Memory Usage: 39.2 MB
 */
public class LargestNumber {
    public String largestNumber(int[] nums) {
        String[] s = new String[nums.length];

        for(int i = 0; i < s.length; i++)
            s[i] = String.valueOf(nums[i]);

        Arrays.sort(s, (e1, e2) -> {
            String a1 = e1 + e2;
            String a2 = e2 + e1;
            return a2.compareTo(a1);
        });

        String answer = "";

        for(int i = 0; i < s.length; i++){
            String a = s[i];

            if(a.equals("0") && answer.equals(""))
                continue;
            answer += a;
        }
        return answer.equals("") ? "0" : answer;
    }
}

File: 10주차/LeetCode_845_Longest_Mountain_in_Array/dongjun.py
'''
1. 문제접근
    투포인터로 풀었습니다.

2. 성능
    Runtime: 164ms
    Memory Usage: 14.9MB
    Time Complexity: O(N)
'''

class Solution:
    def longestMountain(self, A: List[int]) -> int:
        
        i = 0
        answer = 0
        while i < len(A) - 1:
            
            j = i + 1
            visited_top = False
            
            while j < len(A):
                if A[j] > A[j-1]:
                    if visited_top:
                        break
                    
                if A[j] < A[j-1]:
                    if not visited_top:
                        if j - i == 1:
                            break
                        visited_top = True
                
                if A[j] == A[j-1]:
                    break
                j += 1

            if visited_top:
                if j - i > 2:
                    answer = max(answer, j - i)
                
                i = j - 1
                continue
            i = j
            
        return answer
